### PR TITLE
JS Basic and OOP Challenge Revision

### PR DESCRIPTION
**Pull request by dianprsty**
Hello @ianriizky , Thanks for your feedback in my previous pull request #6 .

In this pull request, I already refactor my code and follow your suggestion.
So, All features and functionality is same as my previous pull request #6 but with cleaner code

Here is few general changes that I made in the code
- remove unnecessary comment
- add line break between some line of codes
- make all function's and method's JSDoc have a description
- change variable name to simple yet understandable
- remove some unnecessary else statement
- group some of code to a function to make main method clean
- etc

### CODE REVIEW

File: js/oop/FizzBuzz.js [hunk 1]
```diff
@@ -11,6 +11,16 @@ export class FizzBuzz {
...
 
+  /**
+   * Create an array filled by "fizz", "buzz", or "fizz buzz" based on
+   * this requirement below (highest priority from above).
+   *
+   * Muliples of 4 or 7: "fizz buzz"
+   * Odd sequence: "fizz"
+   * Even sequence: "buzz"
+   *
+   * @returns {Array}
```
Comment (ianriizky) [inline js/oop/FizzBuzz.js:22]: See my review at https://github.com/ngodingbang/rwid-git/pull/12#discussion_r1311503258.

File: js/oop/FizzBuzz.js [hunk 2]
```diff
@@ -26,6 +36,40 @@ export class FizzBuzz {
 
     return result;
   }
+
+  /**
+   * Create string of html code that contain the result of fizz buzz item
```
Comment (ianriizky) [inline js/oop/FizzBuzz.js:41]: See my review at https://github.com/ngodingbang/rwid-git/pull/12#discussion_r1311498400.

File: js/oop/FizzBuzz.js [hunk 3]
```diff
@@ -26,6 +36,40 @@ export class FizzBuzz {
...
+   *
+   * @param {Array} array
```
Comment (ianriizky) [inline js/oop/FizzBuzz.js:43]: See my review at https://github.com/ngodingbang/rwid-git/pull/12#discussion_r1311503258.

File: js/oop/Palindrome.js [hunk 4]
```diff
@@ -11,7 +11,79 @@ export class Palindrome {
     this.value = parseString(value);
   }
 
-  // write your code here
+  /**
+   * Determine whether the given value is a palindrome or not.
+   *
+   * @returns {bool}
```
Comment (ianriizky) [inline js/oop/Palindrome.js:17]: It's "boolean", not "bool".

File: js/oop/Palindrome.js [hunk 5]
```diff
@@ -11,7 +11,79 @@ export class Palindrome {
...
+  isPalindromeUsingReverse() {
+    let reversedValue = this.value.split("").reverse().join("");
+
+    return this.value == reversedValue;
+  }
+
+  /**
+   * Determine whether the given value is a palindrome or not.
+   *
+   * @returns {bool}
```
Comment (ianriizky) [inline js/oop/Palindrome.js:28]: See my review at https://github.com/ngodingbang/rwid-git/pull/12#discussion_r1311525926.

File: js/oop/Palindrome.js [hunk 6]
```diff
@@ -11,7 +11,79 @@ export class Palindrome {
...
+
+    return this.value == reversedValue;
+  }
+
+  /**
+   * Determine whether the given value is a palindrome or not.
+   *
+   * @param {string} reversedValue
+   * @param {number} index
+   * @returns {bool}
```
Comment (ianriizky) [inline js/oop/Palindrome.js:45]: See my review at https://github.com/ngodingbang/rwid-git/pull/12#discussion_r1311525926.

**Review by ianriizky — APPROVED** [review]
LGTM. But you need to check some of the reviews as you continue to learn the next course, thank you.

**Comment by ngodingbang** [comment]
This pull request has been closed with certain reviews here: https://github.com/ngodingbang/rwid-git/pull/12#pullrequestreview-1604557507.

**Pull request closed without merging**